pkg/cloudprovider/vsphere/instances.go

```diff
@@ -55,13 +55,18 @@ func (i *instances) NodeAddresses(ctx context.Context, nodeName types.NodeName)
 		return node.NodeAddresses, nil
 	}
 
-	if err := i.nodeManager.DiscoverNode(string(nodeName), FindVMByName); err != nil {
-		glog.V(2).Infof("instances.NodeAddresses() could NOT FIND node %s err: %v", string(nodeName), err)
-		return []v1.NodeAddress{}, ErrNodeNotFound
+	if err := i.nodeManager.DiscoverNode(string(nodeName), FindVMByName); err == nil {
+		glog.V(2).Info("instances.NodeAddresses() FOUND with ", string(nodeName))
+		if i.nodeManager.nodeNameMap[string(nodeName)] == nil {
+			glog.Errorln("DiscoverNode succeeded, but CACHE missed.")
```

Comment:
I would group these two log lines together, it may overlap with other log lines get confusing. Also would add the node name in there somehow so it's clear which nodes need to be checked for case sensitivity

Reply:
Yup makes sense